pkg/connectors/highlevel_client.go

```diff
@@ -37,16 +39,24 @@ type HighLevelClient interface {
 	SetParallelism(value int)
 	SetBasicAuth(username string, password string)
 	SetHeader(name string, value string)
+	SetPauseBeforeDeploy(pauseBeforeDeploy bool)
+	SetLogFormatter(formatter logrus.Formatter)
```

Comment:
🌹 I'm wondering if this should be part of the public interface, or rather a parameter of `NewClient()` when creating concrete implementation of `Client`.

Reply:
The way I understood/see it is as long as it's not a required parameter (read `url`) it's not provided as a parameter to `NewClient` but can be defined through the public interface.

In particular, those 2 parameters are for now, specific to the `deploy` method, not even common to all others, hence better to be isolated _for now_ in the interface.